components/automate-ui/src/app/pages/team-details/team-details.component.ts

```diff
@@ -135,38 +138,39 @@ export class TeamDetailsComponent implements OnInit, OnDestroy {
         this.isMinorV1 = version === 'v1';
       });
 
-    this.store.select(assignableProjects)
-      .subscribe((assignable: ProjectsFilterOption[]) => {
-        // compiles a list of all available projects
+    combineLatest([
+      this.store.select(assignableProjects),
+      this.store.select(optionsLoadingStatus),
+      this.store.select(v1TeamFromRoute),
+      this.store.select(v2TeamFromRoute),
+      this.store.select(iamMajorVersion)
+    ]).pipe(
+      takeUntil(this.isDestroyed),
+      map(([assignable, pStatus, v1Team, v2Team, major]:
+        [ProjectsFilterOption[], EntityStatus, Team, Team, IAMMajorVersion]) => {
+        this.isMajorV1 = major === 'v1';
+        return [assignable, pStatus, (this.isMajorV1 ? v1Team : v2Team)];
+      }),
+      filter(([_assignable, pStatus, team]:
+        [ProjectsFilterOption[], EntityStatus, Team]) =>
+        pStatus !== EntityStatus.loading && !!team
+      ),
+      map(([assignable, _pStatus, team]) => {
+        this.team = team;
+
         assignable
-          // don't override projects that were already part of the team
-          .filter(p => !this.projects[p.value])
           .forEach(({ value: id, label: name, type: stringType }) => {
             const type = <IAMType>stringType;
             // type and status are unused
-            this.projects[id] = { id, name, type, status: 'NO_RULES', checked: false };
+            this.projects[id] = {
+              id, name, type, status: 'NO_RULES', checked: this.team.projects.includes(id)
+            };
           });
-      });
-
-    this.store.select(iamMajorVersion)
-      .pipe(
-        takeUntil(this.isDestroyed),
-        filter(identity)
-        )
-      .subscribe((version) => {
-        this.isMajorV1 = version === 'v1';
-
-        // Triggered every time the team is updated.
-        if (this.isMajorV1) {
-          this.store.select(v1TeamFromRoute)
-            .pipe(filter(identity), takeUntil(this.isDestroyed))
-            .subscribe(this.getTeamDependentData.bind(this));
-        } else {
-          this.store.select(v2TeamFromRoute)
-            .pipe(filter(identity), takeUntil(this.isDestroyed))
-            .subscribe(this.getTeamDependentData.bind(this));
-        }
-      });
+        this.updateNameForm.controls.name.setValue(this.team.name);
+        this.store.dispatch(new GetTeamUsers({ id: this.teamId }));
+        this.store.dispatch(new GetUsers());
+      }))
+      .subscribe();
```

Comment:
Uhm... couldn't we do the last `map()`'s contents in the `subscribe()`? This looks a bit odd 😉 but it might just be me.